test/Shared/PlatformHelpers.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+#pragma warning disable SA1649 // File name should match first type name
+#pragma warning disable SA1402 // File may only contain a single type
+
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace OpenTelemetry.Tests;
+
+internal enum TestPlatform
+{
+    Unknown = 0,
+    Windows = 1,
+    Linux = 2,
+}
```

Comment:
Why not add MacOS to this list as well? It is needed in some resource detectors, such as os and host.

Reply:
Added: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/2120/commits/9368bedccfcd19569c30892fdacd47c8ebd4cea2